[firebase_remote_config]: firebase_remote_config.lib: error LNK2001: unresolved external symbol __imp__CrtDbgReport

### Is there an existing issue for this?

- [x] I have searched the existing issues.

### Which plugins are affected?

Remote Config

### Which platforms are affected?

Windows

### Description

Hello, our team building flutterfire as a part of testing to detect MSVC regressions. And recently we get the below errors in our daily testing, can you take a look? Thanks.

### Reproducing the issue

1. git clone https://github.com/firebase/flutterfire C:\gitP\firebase\flutterfire
2. download https://storage.googleapis.com/flutter_infra_release/releases/stable/windows/flutter_windows_3.41.3-stable.zip to C:\gitP\firebase\flutterfire\tools
3. set PATH=C:\gitP\firebase\flutterfire\tools\flutter\bin;%PATH%
4. dart pub cache clean --force
5. flutter update-packages --force-upgrade
6. cd /d C:\gitP\firebase\flutterfire\tests\windows\
7. flutter pub get
8. flutter build windows --verbose

### Firebase Core version

latest source code

### Flutter Version

3.41.3

### Relevant Log Output

```shell
C:\gitP\firebase\flutterfire\tests\build\windows\x64\runner>link.exe /ERRORREPORT:QUEUE /OUT:"C:\gitP\firebase\flutterfire\tests\build\windows\x64\runner\Release\tests.exe" /INCREMENTAL:NO /NOLOGO ..\flutter\Release\flutter_wrapper_app.lib dwmapi.lib ..\plugins\firebase_auth\Release\firebase_auth_plugin.lib ..\plugins\firebase_core\Release\firebase_core_plugin.lib ..\plugins\firebase_remote_config\Release\firebase_remote_config_plugin.lib ..\plugins\firebase_storage\Release\firebase_storage_plugin.lib ..\extracted\firebase_cpp_sdk_windows\libs\windows\VS2019\MD\x64\Release\firebase_auth.lib ..\extracted\firebase_cpp_sdk_windows\libs\windows\VS2019\MD\x64\Debug\firebase_remote_config.lib ..\plugins\firebase_core\Release\firebase_core_plugin.lib ..\extracted\firebase_cpp_sdk_windows\libs\windows\VS2019\MD\x64\Release\firebase_app.lib advapi32.lib ws2_32.lib crypt32.lib rpcrt4.lib ole32.lib icu.lib ..\flutter\Release\flutter_wrapper_plugin.lib C:\gitP\firebase\flutterfire\tests\windows\flutter\ephemeral\flutter_windows.dll.lib ..\extracted\firebase_cpp_sdk_windows\libs\windows\VS2019\MD\x64\Release\firebase_storage.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTUAC:"level='asInvoker' uiAccess='false'" /manifest:embed /manifestinput:C:\gitP\firebase\flutterfire\tests\windows\runner\runner.exe.manifest /PDB:"C:/gitP/firebase/flutterfire/tests/build/windows/x64/runner/Release/tests.pdb" /SUBSYSTEM:WINDOWS /TLBID:1 /DYNAMICBASE /NXCOMPAT /IMPLIB:"C:/gitP/firebase/flutterfire/tests/build/windows/x64/runner/Release/tests.lib" /MACHINE:X64  /machine:x64 tests.dir\Release\Runner.res tests.dir\Release\flutter_window.obj tests.dir\Release\main.obj tests.dir\Release\utils.obj tests.dir\Release\win32_window.obj tests.dir\Release\generated_plugin_registrant.obj
   Creating library C:/gitP/firebase/flutterfire/tests/build/windows/x64/runner/Release/tests.lib and object C:/gitP/firebase/flutterfire/tests/build/windows/x64/runner/Release/tests.exp
firebase_remote_config.lib(03798d6ab117d6708bd984671be6866a_firebase_remote_config.dir_Debug_remote_config_response.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport
firebase_remote_config.lib(03798d6ab117d6708bd984671be6866a_firebase_remote_config.dir_Debug_config_data.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport
firebase_remote_config.lib(03798d6ab117d6708bd984671be6866a_firebase_remote_config.dir_Debug_file_manager.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport
firebase_remote_config.lib(03798d6ab117d6708bd984671be6866a_firebase_remote_config.dir_Debug_rest.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport
firebase_remote_config.lib(03798d6ab117d6708bd984671be6866a_firebase_remote_config.dir_Debug_remote_config_request.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport
firebase_remote_config.lib(03798d6ab117d6708bd984671be6866a_firebase_remote_config.dir_Debug_common.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport
firebase_remote_config.lib(03798d6ab117d6708bd984671be6866a_firebase_remote_config.dir_Debug_remote_config.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport
firebase_remote_config.lib(03798d6ab117d6708bd984671be6866a_firebase_remote_config.dir_Debug_remote_config_desktop.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport
firebase_remote_config.lib(03798d6ab117d6708bd984671be6866a_firebase_remote_config.dir_Debug_metadata.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport
firebase_remote_config.lib(03798d6ab117d6708bd984671be6866a_firebase_remote_config.dir_Debug_remote_config_response.obj) : error LNK2001: unresolved external symbol __imp__calloc_dbg
firebase_remote_config.lib(03798d6ab117d6708bd984671be6866a_firebase_remote_config.dir_Debug_config_data.obj) : error LNK2001: unresolved external symbol __imp__calloc_dbg
firebase_remote_config.lib(03798d6ab117d6708bd984671be6866a_firebase_remote_config.dir_Debug_file_manager.obj) : error LNK2001: unresolved external symbol __imp__calloc_dbg
firebase_remote_config.lib(03798d6ab117d6708bd984671be6866a_firebase_remote_config.dir_Debug_rest.obj) : error LNK2001: unresolved external symbol __imp__calloc_dbg
firebase_remote_config.lib(03798d6ab117d6708bd984671be6866a_firebase_remote_config.dir_Debug_remote_config_request.obj) : error LNK2001: unresolved external symbol __imp__calloc_dbg
firebase_remote_config.lib(03798d6ab117d6708bd984671be6866a_firebase_remote_config.dir_Debug_config_update_listener_registration.obj) : error LNK2001: unresolved external symbol __imp__calloc_dbg
firebase_remote_config.lib(03798d6ab117d6708bd984671be6866a_firebase_remote_config.dir_Debug_remote_config.obj) : error LNK2001: unresolved external symbol __imp__calloc_dbg
firebase_remote_config.lib(03798d6ab117d6708bd984671be6866a_firebase_remote_config.dir_Debug_remote_config_desktop.obj) : error LNK2001: unresolved external symbol __imp__calloc_dbg
firebase_remote_config.lib(03798d6ab117d6708bd984671be6866a_firebase_remote_config.dir_Debug_metadata.obj) : error LNK2001: unresolved external symbol __imp__calloc_dbg
firebase_remote_config.lib(03798d6ab117d6708bd984671be6866a_firebase_remote_config.dir_Debug_file_manager.obj) : error LNK2019: unresolved external symbol "__declspec(dllimport) public: static void * __cdecl std::_Crt_new_delete::operator new(unsigned __int64)" (__imp_??2_Crt_new_delete@std@@SAPEAX_K@Z) referenced in function "public: __cdecl std::wstring_convert<class std::codecvt_utf8_utf16<wchar_t,1114111,0>,wchar_t,class std::allocator<wchar_t>,class std::allocator<char> >::wstring_convert<class std::codecvt_utf8_utf16<wchar_t,1114111,0>,wchar_t,class std::allocator<wchar_t>,class std::allocator<char> >(void)" (??0?$wstring_convert@V?$codecvt_utf8_utf16@_W$0BAPPPP@$0A@@std@@_WV?$allocator@_W@2@V?$allocator@D@2@@std@@QEAA@XZ)
firebase_remote_config.lib(03798d6ab117d6708bd984671be6866a_firebase_remote_config.dir_Debug_file_manager.obj) : error LNK2019: unresolved external symbol "__declspec(dllimport) public: static void __cdecl std::_Crt_new_delete::operator delete(void *)" (__imp_??3_Crt_new_delete@std@@SAXPEAX@Z) referenced in function "int `public: __cdecl std::wstring_convert<class std::codecvt_utf8_utf16<wchar_t,1114111,0>,wchar_t,class std::allocator<wchar_t>,class std::allocator<char> >::wstring_convert<class std::codecvt_utf8_utf16<wchar_t,1114111,0>,wchar_t,class std::allocator<wchar_t>,class std::allocator<char> >(void)'::`1'::dtor$3" (?dtor$3@?0???0?$wstring_convert@V?$codecvt_utf8_utf16@_W$0BAPPPP@$0A@@std@@_WV?$allocator@_W@2@V?$allocator@D@2@@std@@QEAA@XZ@4HA)
C:\gitP\firebase\flutterfire\tests\build\windows\x64\runner\Release\tests.exe : fatal error LNK1120: 4 unresolved externals
```

### Flutter dependencies

<!--- Look below for instructions on how to share your Flutter Dependencies. --->

<details>
<summary>Expand <code>Flutter dependencies</code> snippet</summary>
<br>

```yaml

Replace this line with the contents of your `flutter pub deps -- --style=compact`.

```

</details>


### Additional context and comments

For "..\extracted\firebase_cpp_sdk_windows\libs\windows\VS2019\MD\x64\Debug\firebase_remote_config.lib", change "Debug" to "Release" can resolve this issue.

--- a/.github/workflows/windows.yaml
+++ b/.github/workflows/windows.yaml
@@ -53,6 +53,8 @@ jobs:
       - name: "Install Tools"
         run: |
           npm install -g firebase-tools
+      - name: "Build Windows (Release)"
+        run: cd tests && flutter build windows --release
       - name: Start Firebase Emulator and run tests
         run: cd ./.github/workflows/scripts && firebase emulators:exec --project flutterfire-e2e-tests "cd ../../../tests && flutter test .\integration_test\e2e_test.dart -d windows"
 

--- a/packages/firebase_core/firebase_core/windows/CMakeLists.txt
+++ b/packages/firebase_core/firebase_core/windows/CMakeLists.txt
@@ -122,7 +122,7 @@ add_subdirectory(${FIREBASE_CPP_SDK_DIR} bin/ EXCLUDE_FROM_ALL)
 target_include_directories(${PLUGIN_NAME} INTERFACE
   "${FIREBASE_CPP_SDK_DIR}/include")
 
-set(FIREBASE_RELEASE_PATH_LIBS firebase_app firebase_auth firebase_storage firebase_firestore)
+set(FIREBASE_RELEASE_PATH_LIBS firebase_app firebase_auth firebase_remote_config firebase_storage firebase_firestore)
 foreach(firebase_lib IN ITEMS ${FIREBASE_RELEASE_PATH_LIBS})
     get_target_property(firebase_lib_path ${firebase_lib} IMPORTED_LOCATION)
     string(REPLACE "Debug" "Release" firebase_lib_release_path ${firebase_lib_path})

